elfi/methods/bo/acquisition.py

```diff
@@ -244,6 +246,171 @@ def evaluate_gradient(self, x, t=None):
+        t : int, optional
+            Current iteration, (unused).
+
+        Returns
+        -------
+        array_like
+            Coordinates of the yielded acquisition points.
+
+        """
+        logger.debug('Acquiring the next batch of %d values', n)
+        gp = self.model
+
+        # Updating the discrepancy threshold.
+        self.eps = np.percentile(gp.Y, self.quantile_eps * 100)
+
+        def _negate_eval(theta):
+            return -self.evaluate(theta)
+
+        def _negate_eval_grad(theta):
+            return -self.evaluate_gradient(theta)
+
+        # Obtaining the location where the variance is maximised.
+        theta_max, _ = minimize(_negate_eval,
+                                gp.bounds,
+                                _negate_eval_grad,
+                                self.prior,
+                                self.n_inits,
+                                self.max_opt_iters,
+                                random_state=self.random_state)
+
+        # Using the same location for all points in theta batch.
+        batch_theta = np.tile(theta_max, (n, 1))
+
+        return batch_theta
+
+    def evaluate(self, theta_new, t=None):
+        """Evaluate the acquisition function at the location theta_new.
+
+        The rationale of the MaxVar acquisition is based on maximising this evaluation function.
+
+        Parameters
+        ----------
+        theta_new : array_like
+            Evaluation coordinates.
+        t : int, optional
+            Current iteration, (unused).
+
+        Returns
+        -------
+        array_like
+            Variance of the approximate posterior.
+
+        """
+        mean, var = self.model.predict(theta_new, noiseless=True)
+        sigma_n = self.model.noise
+
+        # Using the cdf of Skewnorm to avoid explicit Owen's T computation.
+        a = np.sqrt(sigma_n) / np.sqrt(sigma_n + 2. * var)  # Skewness.
+        scale = np.sqrt(sigma_n + var)
+        phi_skew = ss.skewnorm.cdf(self.eps, a, loc=mean, scale=scale)
+        phi_norm = ss.norm.cdf(self.eps, loc=mean, scale=scale)
+        var_p_a = 2 * phi_norm - phi_skew - phi_norm**2
+
+        val_prior = self.prior.pdf(theta_new).ravel()[:, np.newaxis]
+
+        var_approx_posterior = val_prior**2 * var_p_a
+        return var_approx_posterior
+
+    def evaluate_gradient(self, theta_new, t=None):
+        """Evaluate the acquisition function's gradient at the location theta_new.
+
+        Parameters
+        ----------
+        theta_new : array_like
+            Evaluation coordinates.
+        t : int, optional
+            Current iteration, (unused).
+
+        Returns
+        -------
+        array_like
+            Gradient of the variance of the approximate posterior
+
+        """
+        phi = ss.norm.cdf
+        mean, var = self.model.predict(theta_new, noiseless=True)
+        grad_mean, grad_var = self.model.predictive_gradients(theta_new)
+        noise = self.model.noise
```

Comment:
While checking randmaxvar, realised that sigma_n is usually used to refer to standard deviation and not variance as is now done here. -> Rename sigma_n as sigma2_n everywhere or, altenatively, remove "^2" on line 260 to avoid potential confusion?

Reply:
Went with the latter option and removed "^2"s